qiskit/providers/aer/backends/cluster/clusterjobset.py

```diff
@@ -0,0 +1,236 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A set of jobs being managed by the :class:`AerClusterManager`."""
+
+from typing import List, Optional, Union, Any, Tuple
+from concurrent import futures
+import time
+import logging
+import uuid
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.pulse import Schedule
+from qiskit.qobj import QasmQobj
+from qiskit.providers.jobstatus import JobStatus
+from qiskit.result import Result
+
+from ..aerbackend import AerBackend
+from .clusterjob import CJob
+from .clusterresults import CResults
+from .exceptions import AerClusterTimeoutError, AerClusterJobNotFound
+from .utils import requires_submit
+
+logger = logging.getLogger(__name__)
+
+
+class JobSet:
+    """A set of cluster jobs.
+
+    An instance of this class is returned when you submit experiments using
+    :meth:`AerClusterManager.run()`.
+    It provides methods that allow you to interact
+    with the jobs as a single entity. For example, you can retrieve the results
+    for all of the jobs using :meth:`results()` and cancel all jobs using
+    :meth:`cancel()`.
+    """
+
+    def __init__(self, experiments: List[QasmQobj], name: Optional[str] = None):
+        """JobSet constructor.
+
+        Args:
+            experiments: List[QasmQobjs] to execute.
+            name: Name for this set of jobs.
+        """
+        self._name = name or str(uuid.uuid4())
+        self._experiments = experiments
+
+        # Used for caching
+        self._future = None
+        self._futures = []
+        self._results = None
+        self._error_msg = None
+
+    def run(self,
+            executor: futures.Executor,
+            backend: AerBackend,
+            *run_args: Any,
+            **run_kwargs: Any) -> None:
+        """Execute this set of jobs on an executor.
+
+        Args:
+            executor: The executor used to submit jobs asynchronously.
+            backend: The backend to run the jobs on.
+            *run_args: Positional arguments passed through to backend.run.
+            **run_kwargs: Keyword arguments passed through to backend.run.
+
+        Raises:
+            RuntimeError: If the jobs were already submitted.
+        """
+        if self._futures:
+            raise RuntimeError(
+                'The jobs for this managed job set have already been submitted.')
+
+        self._future = True
+        total_jobs = len(self._experiments)
+        for i, exp in enumerate(self._experiments):
+            cjob = CJob(backend, exp, *run_args, **run_kwargs)
+            cjob.submit(executor=executor)
+            logger.debug("Job %s submitted", i + 1)
+            self._futures.append(cjob)
+
+    @requires_submit
+    def statuses(self) -> List[Union[JobStatus]]:
```

Comment:
There should be a regular `status` method for job that is a single value that combines the subjobs (ie RUNNING if any sub jobs are still running, DONE if all done, etc)

Then you can have a method like this one for returning status of sub-jobs. I don't particularly like the name `statuses`, maybe something like `worker_status` that can take a worker index and return a single status, or if none return the list for all workers

```suggestion
    def worker_status(self, worker: Union[int, Iterable[int]]) -> Union[JobStatus, List[JobStatus]]:
```